Core/include/Acts/Utilities/VectorHelpers.hpp

```diff
@@ -119,7 +120,11 @@ double eta(const Eigen::MatrixBase<Derived>& v) noexcept {
     assert(v.rows() == 3 && "Eta function not valid for non-3D vectors.");
   }
 
-  return std::atanh(v[2] / v.norm());
+  if (v[0] == 0. && v[1] == 0.) {
+    return std::copysign(std::numeric_limits<double>::infinity(), v[2]);
+  } else {
+    return std::asinh(v[2] / std::sqrt(v[0] * v[0] + v[1] * v[1]));
```

Comment:
```suggestion
    return std::asinh(v[2] / perp(v));
```

Reply:
Hi @andiwand, thank you for the suggestion, I've changed the code.